pkg/istio-agent/agent.go

```diff
@@ -613,6 +613,10 @@ func (a *Agent) generateGRPCBootstrap(credentialSocketExists bool) error {
 		return fmt.Errorf("failed generating node metadata: %v", err)
 	}
 
+	// GRPC bootstrap requires this. Original implementation injected this via env variable, but
+	// this interfere with envoy, we should be able to use both envoy for TCP/HTTP and proxyless.
+	node.Metadata.Generator = "grpc"
```

Comment:
do we want to only set it if unset? I don't know why, but hypothetically someone might want to override it?